src/crate/client/cursor.py

```diff
@@ -53,8 +54,37 @@ def execute(self, sql, parameters=None, bulk_parameters=None):
         self._result = self.connection.client.sql(sql, parameters,
                                                   bulk_parameters)
         if "rows" in self._result:
+            if "col_types" in self._result:
+                rows_to_convert = self._get_rows_to_convert_to_date(self._result["col_types"])
+                for flag in rows_to_convert:
+                    if flag:
+                        t_rows = (row for row in self._result["rows"])
```

Comment:
This will probably still load the full result into memory, right?

I was thinking about swapping in the generator right in between of `self.rows = iter(self._result["rows"])`, which is then responsible for converting between data types (timestamp epoch integer to Python datetime object).

In that manner, the pre-computation using `_get_rows_to_convert_to_date()` might become obsolete.